### PR TITLE
test: add simulated power management failures in testing mode

### PR DESCRIPTION
**Pull request by smira**
This injects power management failures in Sfyra (under test). By default
failures are not injected.

Two class of failures are supported:

* explicit (when client returns an error)
* silent (when client doesn't do the job, but error is not returned)

Signed-off-by: Andrey Smirnov <smirnov.andrey@gmail.com>

### CODE REVIEW

**Review by rsmitty — DISMISSED** [review]

**Review by andrewrynhard — APPROVED** [review]

**Comment by smira** [comment]
I don't want to merge this PR just yet until we have a better stability of Sidero, as this might introduce some failures into Sfyra we would have to dig into to understand if it's introduced failure or what not. should be easy enough to merge it later

File: sfyra/cluster.yaml [hunk 1]
```diff
@@ -0,0 +1,108 @@
+apiVersion: cluster.x-k8s.io/v1alpha3
```
Comment (smira) [inline sfyra/cluster.yaml:1]: TODO: remove this

**Pull request merged**